File: cpp/example/src/DuplicateZeros/duplicateZeros.cpp
#include "duplicateZeros.h"

void duplicateZeros::naive(vector<int>& arr){
    int i=0,origLen=arr.size();
    vector<int>::iterator it = arr.begin();
    while (i<origLen){
        if (arr[i]==0){
            arr.pop_back();
            arr.insert(it+i,1,0);
            i+=1;
        }
        i+=1;
    }
}

File: cs/sample/Sample/DuplicateZeros/DuplicateZeros.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Sample.DuplicateZeros{
    public class duplicateZeros{
        public static void naive(int[] arr){
            int i=0,origLen=arr.Length;
            while (i<origLen){
                if (arr[i]==0){
                    for (int j=origLen-1;j>i+1;j--){
                        arr[j] = arr[j-1];
                    }
                    arr[i+1] = 0;
                    i+=1;
                }
                i+=1;
            }
        }
    }
}
